lib/web/apps.go

```diff
@@ -61,8 +62,24 @@ func (h *Handler) clusterAppsGet(w http.ResponseWriter, r *http.Request, p httpr
 	}
 
 	page, err := apiclient.GetResourcePage[types.AppServerOrSAMLIdPServiceProvider](r.Context(), clt, req)
+
 	if err != nil {
-		return nil, trace.Wrap(err)
+		// If the error returned is due to types.KindAppOrSAMLIdPServiceProvider being unsupported, then fallback to attempting to just fetch types.AppServers.
+		// This is for backwards compatibility with leaf clusters that don't support this new type yet.
+		if strings.Contains(err.Error(), types.KindAppOrSAMLIdPServiceProvider) {
```

Comment:
I wonder if there's a better way than checking the string error message.

@rosstimothy ?